google-cloud-translate/src/main/java/com/google/cloud/translate/TranslateOptions.java

```diff
@@ -135,7 +136,19 @@ public TranslateOptions build() {
 
   private TranslateOptions(Builder builder) {
     super(TranslateFactory.class, TranslateRpcFactory.class, builder, new TranslateDefaults());
-    this.apiKey = builder.apiKey != null ? builder.apiKey : getDefaultApiKey();
+    if (builder.apiKey != null) {
+      this.apiKey = builder.apiKey;
+    } else if (System.getenv(ServiceOptions.CREDENTIAL_ENV_NAME) == null) {
+      this.apiKey = getDefaultApiKey();
+    } else {
```

Comment:
Perhaps the message should change to "API key is unset; Using GOOGLE_APPLICATION_CREDENTIALS" ?

You still need a test for the apiKey set && GOOGLE_APPLICATION_CREDENTIALS set case?  As that can still happen.   Perhaps to warn that the Service account will override it.